UMA-Bots/Configuration-Change/src/agent.ts

```diff
@@ -0,0 +1,56 @@
+import { Finding, HandleTransaction, ethers, Initialize, TransactionEvent, getEthersProvider } from "forta-agent";
+import { getFindingInstance, HUBPOOL_MONITORED_EVENTS, SPOKEPOOL_MONITORED_EVENTS, getMetadata } from "./utils";
+import { NetworkManager } from "forta-agent-tools";
+import { NM_DATA, NetworkDataInterface } from "./network";
+
+const networkManager = new NetworkManager(NM_DATA);
+
+export function provideInitialize(
+  networkManager: NetworkManager<NetworkDataInterface>,
+  provider: ethers.providers.Provider
+): Initialize {
+  return async () => {
+    await networkManager.init(provider);
+  };
+}
+
+export function provideHandleTransaction(
+  monitoredHubPoolEvents: string[],
```

Comment:
So, the idea for this was that the HubPool events are always the same no matter what the chain. However, the SpokePool events that need to be monitored are chain-specific.

I am changing this in the commit "feat: hubPoolEvents in NM" please give it a look @Vxatz. However, the only point to note is that the developer must ensure that he is passing the HubPool events (if he also is passing the HubPool address). In case no HubPool address is passed (i.e., there is no HubPool deployed on that chain), the HubPool events also do not need to be passed. Let me know if you feel this approach is better.

Reply:
It's better imo, yes 👍